docs/advanced-usage.md

```diff
@@ -0,0 +1,290 @@
+# Usage
+- [Selecting Java distribution](#Selecting-Java-distribution)
+  - [Adoptium](#Adoptium)
+  - [Zulu](#Zulu)
+- [Installing custom Java package type](#Installing-custom-Java-package-type)
+- [Installing custom Java architecture](#Installing-custom-Java-architecture)
+- [Installing custom Java distribution from local file](#Installing-Java-from-local-file)
+- [Testing against different Java distributions](#Testing-against-different-Java-distributions)
+- [Testing against different platforms](#Testing-against-different-platforms)
+- [Publishing using Apache Maven](#Publishing-using-Apache-Maven)
+- [Publishing using Gradle](#Publishing-using-Gradle)
+
+See [action.yml](../action.yml) for more details on task inputs.
+
+## Selecting Java distribution
+Input `distribution` is mandatory and should be provided to use action. See [Supported distributions](../README.md#Supported-distributions) for the list of available options.
+
+### Adoptium
+```yaml
+steps:
+- uses: actions/checkout@v2
+- uses: actions/setup-java@v2-preview
+  with:
+    distribution: 'adoptium'
+    java-version: '11'
+- run: java -cp java HelloWorldApp
+```
+
+### Zulu
+```yaml
+steps:
+- uses: actions/checkout@v2
+- uses: actions/setup-java@v2-preview
+  with:
+    distribution: 'zulu'
+    java-version: '11'
+    java-package: jdk # optional (jdk, jre, jdk+fx or jre+fx) - defaults to jdk
+- run: java -cp java HelloWorldApp
+```
+
+## Installing custom Java package type
+```yaml
+steps:
+- uses: actions/checkout@v2
+- uses: actions/setup-java@v2-preview
+  with:
+    distribution: '<distribution>'
+    java-version: '11'
+    java-package: jdk # optional (jdk or jre) - defaults to jdk
+- run: java -cp java HelloWorldApp
+```
+
+
+## Installing custom Java architecture
+
+```yaml
+steps:
+- uses: actions/checkout@v2
+- uses: actions/setup-java@v2-preview
+  with:
+    distribution: '<distribution>'
+    java-version: '11'
+    architecture: x86 # optional - defaults to x64
+- run: java -cp java HelloWorldApp
+```
+
+## Installing Java from local file
+If your use-case requires custom distribution or version that is not provided by action, you can download it manually and then action will take care about installation and caching of this version on VM:
```

Comment:
If your use-case requires a custom distribution or a version that is not provided by setup-java, you can download it manually and setup-java will take care of the installation and caching on the VM: